react/components/Login/index.js

```diff
@@ -0,0 +1,126 @@
+import React, { Component } from 'react'
+import Button from '@vtex/styleguide/lib/Button'
+
+import LoginOptions from './components/LoginOptions'
+import EmailVerification from './components/EmailVerification'
+import CodeConfirmation from './components/CodeConfirmation'
+import ProfileIcon from './images/ProfileIcon'
+
+import './global.css'
+
+const GO_BACK = 'login.go-back'
+
+const CONSTANTS = [
+  {
+    page: 'login-options',
+    titleLable: 'login-options.title',
+    options: ['login-options.email-verification'],
+  },
+  {
+    goBack: GO_BACK,
+    send: 'login.send',
+    next: 2,
+    previous: 0,
+  },
+  {
+    goBack: GO_BACK,
+    send: 'login.send',
+    titleLable: 'login-email-code.title',
+    previous: 1,
+  },
+]
+
+/** Canonical login that calls a mutation to retrieve the authentication token */
+export default class Login extends Component {
+  state = {
+    isMouseOnButton: false,
+    isMouseOnContent: false,
+    step: 0,
+    email: '',
+    code: '',
+    authtoken: '',
+  }
+
+  handleUpdateState = state => {
+    this.setState(state)
+  }
+
+  render() {
+    const {
+      isMouseOnButton,
+      isMouseOnContent,
+      email,
+      code,
+      authtoken,
+    } = this.state
+
+    let render
```

Comment:
I think it would be better if the `CONSTANTS` array were an array of functions that receive the props and render the component based on the `this.state.step`.